### PR TITLE
Search all regions when none is specified

### PR DESCRIPTION
**Pull request by brianstorti**
If we don't pass a `region_code` to the query, instead of using "US" by default we are now aggregating the results for all the supported regions.

This is useful when we want to find all the phone number matches, no matter the region (e.g. for redaction).

### Caveats

As a given number can be a match for several different regions, we are returning just unique results. For example, the number `07511364301` is a match for both UK and China (and several other regions), so instead of returning this same match multiple times, we return just the first one.

This can be a bit weird if we are also requesting things like `country_code` and `international_format`, as the same number can have multiple different international formats, and we will return just the first one.

### CODE REVIEW

File: lib/phone_utility.rb [hunk 1]
```diff
@@ -18,8 +18,9 @@ def self.parse(number, region)
     instance.parse(number, region)
   end
 
-  def self.find_numbers(text, default_region)
-    instance.find_numbers(text, default_region)
+  def self.find_numbers(text, default_region = nil)
+    regions = default_region ? [default_region] : instance.get_supported_regions
```
Comment (daemonsy) [inline lib/phone_utility.rb]: Hi hi, how does the performance of this call look? Just wondering how large the list of supported regions are. 

Also, what happens when when a number is matched in two regions? There's a chance libphonenumber can return an invalid match (I.e. I really want this to look like a Singapore number, but it isn't a valid one, the library returns a match with `isValid` false

Just some considerations

Comment (brianstorti) [inline lib/phone_utility.rb]: About the performance, the impact is not very noticeable ([gist](https://gist.github.com/brianstorti/8bceb279fe27d5c934a733907aa98397)):

```
Warming up --------------------------------------
         with region     1.000  i/100ms
      without region     1.000  i/100ms
Calculating -------------------------------------
         with region     47.297  (±27.5%) i/s -    403.000  in   9.985090s
      without region     43.539  (±23.0%) i/s -    411.000  in   9.995946s

Comparison:
         with region:       47.3 i/s
      without region:       43.5 i/s - same-ish: difference falls within error
```

Comment (brianstorti) [inline lib/phone_utility.rb]: About the second point, I think this is related to what I described in the `Caveats` section in the description. If we want to check that a number is a valid number for Singapore, we'll need to pass `SG` as the region code.

Comment (daemonsy) [inline lib/phone_utility.rb]: Nice! That's good, thanks, I missed the caveats, missed it while reading on the subway. 

I have a little request, can we add the fact that the matching API can return you invalid results and multiple matches on the query type? Leverage the self documenting property of GQL!

Comment (brianstorti) [inline lib/phone_utility.rb]: Yeah, that's a good idea. Just pushed a description for this field, let me know if you think we should add more information there.

**Pull request merged**